### PR TITLE
Overwrite content in collection when same fullPath

### LINKED ISSUE #436: New file doesn't get added to collections
Bug description:
When adding the same file with the same name in the same `dir` on a collection for some reason it doesn't add the new file.

Expected behavior: either the new file should replace the old one (we are able to tell by the new contentID on the file) or both files should be in the collection

Actual behavior: the new file simply doens't get added to the collection (no errors are returned either). 

```bash
path=/my/path

curl -X POST -H Authorization: Bearer EST599c28b4-fd89-4b44-b579-74af0997a5ecARY http://localhost:3004/collections/create -d { "name": "A new collection", "description": "A new collection test" } -s

coluuid=994c07a2-0f4a-4a36-8893-15c5eb291634

curl -X POST -H Authorization: Bearer EST599c28b4-fd89-4b44-b579-74af0997a5ecARY http://localhost:3004/content/add?coluuid=994c07a2-0f4a-4a36-8893-15c5eb291634&dir=/my/path -F data=@estuary -s

file1=54
# added file1 (54) to estuary

curl -X POST -H Authorization: Bearer EST599c28b4-fd89-4b44-b579-74af0997a5ecARY http://localhost:3004/content/add?coluuid=994c07a2-0f4a-4a36-8893-15c5eb291634&dir=/my/path -F data=@estuary -s

file2=55
# added file2 (55) to estuary

curl -X GET -H Authorization: Bearer EST599c28b4-fd89-4b44-b579-74af0997a5ecARY http://localhost:3004/collections/content?coluuid=994c07a2-0f4a-4a36-8893-15c5eb291634 -s
jq
[
  {
    "id": 54,
    "updatedAt": "2022-09-14T18:30:54.51407519Z",
    "cid": "bafybeiezx5rydlqhyqzlqnc7zz7eawqowyzfjoon6b43dbkvws5rop3ivy",
    "name": "estuary",
    "userId": 1,
    "description": "",
    "size": 193993620,
    "type": 0,
    "active": true,
    "offloaded": false,
    "replication": 6,
    "aggregatedIn": 0,
    "aggregate": false,
    "pinning": false,
    "pinMeta": "",
    "replace": false,
    "origins": "",
    "failed": false,
    "location": "local",
    "dagSplit": false,
    "splitFrom": 0,
    "path": "/my/path/estuary"
  }
]
```

### PR DESCRIPTION
**Pull request by gmelodie**
Now if you specify `overwrite=true` it will replace the file with the same name on the collection, and it'll error out if you try doing so without the flag. 

Fixes #436

### CODE REVIEW

**Comment by en0ma** [comment]
@gmelodie there are multiple places contents are added to a collection. It would be nice to abstract the add content to collection logic (with the override flag) and use it everywhere

**Comment by gmelodie** [comment]
@en0ma done! Can you help testing?

File: api/v1/handlers.go [hunk 1]
```diff
@@ -5098,52 +5031,17 @@ func sanitizePath(p string) (string, error) {
 func (s *apiV1) handleColfsAdd(c echo.Context, u *util.User) error {
 	coluuid := c.QueryParam("coluuid")
 	contid := c.QueryParam("content")
-	npath := c.QueryParam("path")
```
Comment (en0ma) [inline api/v1/handlers.go:5101]: honestly, we should be using `path`. It is the right word for what is being done. Collection API v2 should change `dir` to `path`

**Comment by alvin-reyes** [comment]
Let's fix the conflicts.

**Comment by anjor** [comment]
Oh yes! Let's get this merged please. It will fix one of the rclone tests. See https://github.com/rclone/rclone/issues/5854#issuecomment-1356494926

File: api/v1/handlers.go (hunk 1 above)
Comment (gmelodie) [reply api/v1/handlers.go:5015]: These changes are on the v1 file, not v2

**Comment by anjor** [comment]
Tested
```
anjor@seven ~ $ curl http://localhost:3004/collections/ -H "Authorization: Bearer $(cat ~/.estuary_local)" -XPOST -d '{"name":"overwrite test 5", "description":""}' | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   198  100   153  100    45  33663   9900 --:--:-- --:--:-- --:--:--  193k
{
  "createdAt": "2023-01-04T10:37:32.505977Z",
  "uuid": "f028d521-a575-47ad-9ce1-52b47f6c0e13",
  "name": "overwrite test 5",
  "description": "",
  "userId": 1,
  "cid": ""
}
anjor@seven ~ $ coluuid=f028d521-a575-47ad-9ce1-52b47f6c0e13
anjor@seven ~ $ curl "http://localhost:3004/content/add?coluuid=$coluuid&dir=/test" -H "Authorization: Bearer $(cat ~/.estuary_local)" -H "Content-Type: multipart/form-data" -F "data=@a" | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   703  100   491  100   212  59623  25743 --:--:-- --:--:-- --:--:--  137k
{
  "cid": "bafkreicshyqmt4kwtgdeiqzpkt74wodvo4jnukbiqqnca7xcta7js7qrjm",
  "retrieval_url": "https://dweb.link/ipfs/bafkreicshyqmt4kwtgdeiqzpkt74wodvo4jnukbiqqnca7xcta7js7qrjm",
  "estuary_retrieval_url": "https://api.estuary.tech/gw/ipfs/bafkreicshyqmt4kwtgdeiqzpkt74wodvo4jnukbiqqnca7xcta7js7qrjm",
  "estuaryId": 9278,
  "providers": [
    "/ip4/192.168.86.28/tcp/6744/p2p/12D3KooWG7it68ksNLmK9SjSBEwTM68NHzgaKHUXcm21DhUAuo2k",
    "/ip4/127.0.0.1/tcp/6744/p2p/12D3KooWG7it68ksNLmK9SjSBEwTM68NHzgaKHUXcm21DhUAuo2k"
  ]
}
anjor@seven ~ $ vim
anjor@seven ~ $ curl "http://localhost:3004/content/add?coluuid=$coluuid&dir=/test" -H "Authorization: Bearer $(cat ~/.estuary_local)" -H "Content-Type: multipart/form-data" -F "data=@a" | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   357  100   145  100   212  35696  52191 --:--:-- --:--:-- --:--:--  348k
{
  "error": {
    "code": 400,
    "reason": "ERR_CONTENT_IN_COLLECTION",
    "details": "file already exists in collection, specify 'overwrite=true' to overwrite"
  }
}
anjor@seven ~ $ curl "http://localhost:3004/content/add?coluuid=$coluuid&dir=/test&overwrite=true" -H "Authorization: Bearer $(cat ~/.estuary_local)" -H "Content-Type: multipart/form-data" -F "data=@a" | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   703  100   491  100   212  68527  29588 --:--:-- --:--:-- --:--:--  171k
{
  "cid": "bafkreicshyqmt4kwtgdeiqzpkt74wodvo4jnukbiqqnca7xcta7js7qrjm",
  "retrieval_url": "https://dweb.link/ipfs/bafkreicshyqmt4kwtgdeiqzpkt74wodvo4jnukbiqqnca7xcta7js7qrjm",
  "estuary_retrieval_url": "https://api.estuary.tech/gw/ipfs/bafkreicshyqmt4kwtgdeiqzpkt74wodvo4jnukbiqqnca7xcta7js7qrjm",
  "estuaryId": 9279,
  "providers": [
    "/ip4/192.168.86.28/tcp/6744/p2p/12D3KooWG7it68ksNLmK9SjSBEwTM68NHzgaKHUXcm21DhUAuo2k",
    "/ip4/127.0.0.1/tcp/6744/p2p/12D3KooWG7it68ksNLmK9SjSBEwTM68NHzgaKHUXcm21DhUAuo2k"
  ]
}
anjor@seven ~ $
```

**Comment by anjor** [comment]
Tested
```
anjor@seven ~ $ curl http://localhost:3004/collections/ -H "Authorization: Bearer $(cat ~/.estuary_local)" -XPOST -d '{"name":"overwrite test 5", "description":""}' | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   198  100   153  100    45  33663   9900 --:--:-- --:--:-- --:--:--  193k
{
  "createdAt": "2023-01-04T10:37:32.505977Z",
  "uuid": "f028d521-a575-47ad-9ce1-52b47f6c0e13",
  "name": "overwrite test 5",
  "description": "",
  "userId": 1,
  "cid": ""
}
anjor@seven ~ $ coluuid=f028d521-a575-47ad-9ce1-52b47f6c0e13
anjor@seven ~ $ curl "http://localhost:3004/content/add?coluuid=$coluuid&dir=/test" -H "Authorization: Bearer $(cat ~/.estuary_local)" -H "Content-Type: multipart/form-data" -F "data=@a" | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   703  100   491  100   212  59623  25743 --:--:-- --:--:-- --:--:--  137k
{
  "cid": "bafkreicshyqmt4kwtgdeiqzpkt74wodvo4jnukbiqqnca7xcta7js7qrjm",
  "retrieval_url": "https://dweb.link/ipfs/bafkreicshyqmt4kwtgdeiqzpkt74wodvo4jnukbiqqnca7xcta7js7qrjm",
  "estuary_retrieval_url": "https://api.estuary.tech/gw/ipfs/bafkreicshyqmt4kwtgdeiqzpkt74wodvo4jnukbiqqnca7xcta7js7qrjm",
  "estuaryId": 9278,
  "providers": [
    "/ip4/192.168.86.28/tcp/6744/p2p/12D3KooWG7it68ksNLmK9SjSBEwTM68NHzgaKHUXcm21DhUAuo2k",
    "/ip4/127.0.0.1/tcp/6744/p2p/12D3KooWG7it68ksNLmK9SjSBEwTM68NHzgaKHUXcm21DhUAuo2k"
  ]
}
anjor@seven ~ $ vim
anjor@seven ~ $ curl "http://localhost:3004/content/add?coluuid=$coluuid&dir=/test" -H "Authorization: Bearer $(cat ~/.estuary_local)" -H "Content-Type: multipart/form-data" -F "data=@a" | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   357  100   145  100   212  35696  52191 --:--:-- --:--:-- --:--:--  348k
{
  "error": {
    "code": 400,
    "reason": "ERR_CONTENT_IN_COLLECTION",
    "details": "file already exists in collection, specify 'overwrite=true' to overwrite"
  }
}
anjor@seven ~ $ curl "http://localhost:3004/content/add?coluuid=$coluuid&dir=/test&overwrite=true" -H "Authorization: Bearer $(cat ~/.estuary_local)" -H "Content-Type: multipart/form-data" -F "data=@a" | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   703  100   491  100   212  68527  29588 --:--:-- --:--:-- --:--:--  171k
{
  "cid": "bafkreicshyqmt4kwtgdeiqzpkt74wodvo4jnukbiqqnca7xcta7js7qrjm",
  "retrieval_url": "https://dweb.link/ipfs/bafkreicshyqmt4kwtgdeiqzpkt74wodvo4jnukbiqqnca7xcta7js7qrjm",
  "estuary_retrieval_url": "https://api.estuary.tech/gw/ipfs/bafkreicshyqmt4kwtgdeiqzpkt74wodvo4jnukbiqqnca7xcta7js7qrjm",
  "estuaryId": 9279,
  "providers": [
    "/ip4/192.168.86.28/tcp/6744/p2p/12D3KooWG7it68ksNLmK9SjSBEwTM68NHzgaKHUXcm21DhUAuo2k",
    "/ip4/127.0.0.1/tcp/6744/p2p/12D3KooWG7it68ksNLmK9SjSBEwTM68NHzgaKHUXcm21DhUAuo2k"
  ]
}
anjor@seven ~ $
```

**Review by anjor — APPROVED** [review]

**Pull request merged**